keyless/pepper/common/src/lib.rs

```diff
@@ -25,6 +25,27 @@ where
     hex::decode(s).map_err(D::Error::custom)
 }
 
+/// Custom serialization function to convert Vec<u8> into a hex string with the 0x prefix.
+fn serialize_bytes_to_hex_with_0x<S>(bytes: &Vec<u8>, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let hex_string = format!("0x{}", hex::encode(bytes));
+    serializer.serialize_str(&hex_string)
+}
+
+/// Custom deserialization function to convert a hex string with the 0x prefix back into Vec<u8>.
+fn deserialize_bytes_from_hex_with_0x<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    if !s.starts_with("0x") {
```

Comment:
what about `0X`?

Reply:
I think 0x is right since that how serialization does it.

serialization and deserialization is supposed to 1 to 1.  the domain of deserialization should be the range of serialization